pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/SimpleLoadManagerImpl.java

```diff
@@ -1416,6 +1420,8 @@ public void doNamespaceBundleSplit() throws Exception {
 
     @Override
     public void stop() throws PulsarServerException {
-        // do nothing
+        loadReportCacheZk.shutdown();
+        availableActiveBrokers.shutdown();
+        scheduler.shutdown();
```

Comment:
Do you mean to say that it is already handled? There is actually an issue in that both the `LocalBrokerData` and `LoadReport` caches use the same underlying cache, being `pulsar.getLocalZkCache()`. What this means is that, if we were not to shutdown the caches, the integrity of the Object in the cache is determined by which watch fires *last*. This is the cause of the `ClassCastException`: Even though the scheduler is shutdown, the watch fires for all previously created load managers, deserializing the json to either a `LocalBrokerData` or `LoadReport`. The last one to do so wins, which could cause us to read a `LoadReport` when we want a `LocalBrokerData` or vice-versa.